app/javascript/components/MemberDetailsForm/MemberDetailsForm.js

```diff
@@ -145,6 +145,25 @@ export class MemberDetailsForm extends Component {
     );
   }
 
+  fieldsToDisplay() {
+    return this.props.fields.filter(field => {
+      switch (field.display_mode) {
+        case 'all_members':
+          return true;
+        case 'recognized_members_only':
+          return this.recognizedMemberPresent();
+        case 'new_members_only':
+          return !this.recognizedMemberPresent();
+        default:
+          throw `Unknown display_mode "${field.display_mode}" for field "${field.name}"`;
```

Comment:
We shouldn't throw unless we're handling the exception, since that would cause the entire component to not render unless we handle it at some level (e.g. with an [error boundary](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html)). 

Instead of throwing an exception, perhaps we can filter that field out and log an error, or default to displaying the field (I reckon it's safer to hide the field unless we know for certain that we should show it).

Reply:
Great catch Vince, I'll do what you suggested.